fecgov/fec-cms
# Address polyfills for Internet Explorer (IE)

## Summary
Some tweaks to make our site/app work a little better for those still using outdated browsers (mostly Internet Explorer 10 & 11). This is a test ticket because we're changing the X-UA meta tag on every page of the site and we're changing the build process for every datatable page.

- Resolves #2958 
- Resolves #2870 
- Resolves https://github.com/fecgov/FEC/issues/5751

## Impacted areas of the application
- **Updated our `X-UA-Compatible` meta tag to target `edge` instead of `IE11`** ([Microsoft](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/dev-guides/bg182625(v=vs.85)): "edge mode is the preferred document mode; it represents the highest support for modern standards available to the browser." 
- **Added `require('babel-polyfill');`** to aggregate-totals*.js and datatables-*.js files.
- `package.json` has been updated to include our targeted browsers & versions. The values are pretty standard, though we added IE 10, too.
- Compensated for IE not supporting `document.currentScript` (aggregate-totals-box).

## Screenshots
None

## Related PRs
None

## How to test
- Pull the branch.
- `npm run build`.
- `npm run build-production`.
- `npm run test-single`.
- Any errors?
- Do the JavaScript-powered parts of the site still work in non-IE?
- Do the JavaScript-powered parts of the site work in IE 11? IE 10?
- If there are no errors, it's likely things will work normally with modern browsers. Then it's just up to IE 10 & 11.
High priority JavaScript-powered components to check: datatables, aggregate totals (behind a feature flag).
- Want to make sure we didn't re-break #2424 


## Possible next steps
- Roll the change into develop and/or release.
- Update any other pages that reference polyfills to use the same `require('babel-polyfill');` method.
- Check the backlog for IE-specific bugs and whether this addresses those.
____
